bayes/bayesnet.html

```diff
@@ -7,17 +7,49 @@
         <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
         <script src="/js/viz-lite.js"></script>
         <script>
-            function draw() {
-                $.post('/bayes/parse', $('#bayes').val()).done(function(response) {
-                    var div = $('#rhs');
-                    div.empty();
-                    if (response.substring(0, 5) === 'Error') {
-                        div.append(response);
+            // On load
+            $(function () {
+                var $bayes = $('#bayes');
+                var $rhs = $('#rhs');
+                var hashRegex = /[#&]network(?:=([^&#]*)|&|$)/;
+
+                function draw() {
+                    var network = $bayes.val()
+                    $.post('/bayes/parse', network).done(function(response) {
+                        $rhs.empty();
+                        if (response.substring(0, 5) === 'Error') {
+                            $rhs.append(response);
+                        } else {
+                            $rhs.append(Viz(response));
+                        }
+                    });
+                    saveNetwork(network);
+                }
+
+                function getSavedNetwork() {
+                    var hash = location.hash;
+                    if (hashRegex.test(hash)) {
+                        var encoded = hashRegex.exec(hash)[1];
+                        return window.atob(encoded);
                     } else {
-                        div.append(Viz(response));
+                        return '';
                     }
+                }
+
+                function saveNetwork(network) {
+                    location.hash = '#network=' + window.btoa(network);
+                }
+
+                var initialNetwork = getSavedNetwork();
```

Comment:
By wrapping the entire script tag here inside a jQuery tag, all of this code is called on load and avoids polluting the global namespace. (See [this note from the jQuery docs](https://jquery.com/upgrade-guide/3.0/#deprecated-document-ready-handlers-other-than-jquery-function)). Would adding some comments in this section make it clear enough?